raw/era5_sfc_cisst.cfg

```diff
@@ -28,7 +28,7 @@ stream=oper
 expver=1
 levtype=sfc
 type=an
-year=1979/to/2021
+year=1979/to/2023
 month=01/to/12
 day=01/to/31
```

Comment:
I think we may want to add a feature in weather-dl to not throw errors during this type of config when we attempt to download data for future dates. WDYT? @mahrsee1997, what are your thoughts?

Reply:
Just thinking our loud -- I understand the concerns about handling errors related to future dates in the weather-dl tool. While implementing a feature to check the system's date can help, it may not guarantee accuracy due to potential discrepancies. Currently if we pass such config in the tool, our parser will parse the config successfully & create the shards. But then when we place fetch request to client it will throw error.

Considering the complexity of various config issues that can arise (eg: incorrect param id), it might be more practical to delegate the responsibility of providing correct configurations to the user.

Overall, I agree that having a feature to handle future dates can be beneficial (lets create an issue in weather-tools & someone can pick that), but we should also consider the broader range of config issues. Delegating the responsibility to the user ensures they are aware of potential inconsistencies and can provide accurate config accordingly.

Reply:
That's very reasonable and principled, Rahul. It's better to be correct than polite. In that case, I think our focus should be on a programmatic way to update these configs to designate the correct new data to download.